### ISSUE TITLE
Spring Boot 3.1.1 doesn't find custom ObjectMapper

### ISSUE THREAD

**Issue by z0mb1ek** [vladmihalcea/hypersistence-utils#644]
I tested two version of spring boot: 3.1.0 and 3.1.1 with `io.hypersistence:hypersistence-utils-hibernate-62:3.5.1` and setting properties in application.yml like this:

```
spring:
  jpa:
    properties:
      hypersistence:
        utils:
          jackson:
            object:
              mapper: tech.livecom.libraries.utils.hibernate.CustomObjectMapperSupplier
```

In version 3.1.1 it is not working, but works in 3.1.0

**Comment by z0mb1ek** [comment]
I found that with `org.springframework:spring-orm:6.0.9` it starts working, but boot 3.1.1 uses `org.springframework:spring-orm:6.0.10`

**Comment by vladmihalcea** [comment]
Maybe it's a Spring issue. Try to debug it and see what's causing the problem

**Comment by z0mb1ek** [comment]
I see that this constructor doesn't work: 
```
public JsonBinaryType(org.hibernate.type.spi.TypeBootstrapContext typeBootstrapContext) {
        this(new JsonConfiguration(typeBootstrapContext.getConfigurationSettings()));
    }
```

**Comment by vladmihalcea** [comment]
Then it's a Hibernate or Spring issue since that constructor should have been called. You should investigate what caused the problem and open a ticket in the project that caused it.

**Comment by kborghs** [comment]
I came across the same issue. 
When debugging `JsonConfiguration`, I saw that the properties were loaded in `Configuration.load()`. Properties are only collected via `application.properties`, `hypersistence-utils.properties` and some deprecated paths, NOT via `properties.yml`. I haven't found anything in the code that reads the properties from yaml or gets them from spring.

The workaround I have used was to create `hypersistence-utils.properties` and add the custom ObjectMapper with the property `hypersistence.utils.jackson.object.mapper`
